packages/astro/src/core/config/schema.ts

```diff
@@ -614,6 +623,85 @@ export const AstroConfigSchema = z.object({
 					}
 					return svgConfig;
 				}),
+
+			fonts: z
+				.object({
+					providers: z.array(fontProviderSchema).optional(),
+					families: z.array(
+						z.union([
+							// Shorthand
+							z
+								.string()
+								.transform(
+									(name): FontFamilyAttributes => ({
+										name,
+										provider: GOOGLE_PROVIDER_NAME,
+									}),
+								),
+							localFontFamilySchema,
+							commonFontFamilySchema,
+						]),
+					),
+				})
+				.strict()
+				.optional()
+				.superRefine((fonts, ctx) => {
+					if (!fonts) {
+						return;
+					}
+					const providersNames = [
+						...BUILTIN_PROVIDERS,
+						...(fonts.providers ?? []).map((provider) => provider.name),
+					];
+					const visited = new Map<string, 'name' | 'as'>();
+
+					for (let i = 0; i < fonts.families.length; i++) {
+						const family = fonts.families[i];
+
+						// Check for invalid providers
+						if (!providersNames.includes(family.provider)) {
+							ctx.addIssue({
+								code: z.ZodIssueCode.custom,
+								message: `Invalid provider "${family.provider}". Please use of the following: ${providersNames.map((name) => `"${name}"`).join(', ')}`,
+								path: ['families', i, 'provider'],
+							});
+						}
+
+						if (family.as) {
+							if (!VALID_CHAR_RE.test(family.as)) {
+								ctx.addIssue({
+									code: z.ZodIssueCode.custom,
+									message: `**as** property "${family.as}" contains invalid characters for CSS variable generation. Only letters, numbers, spaces, underscores and colons are allowed.`,
+									path: ['families', i, 'as'],
+								});
+							}
+						} else if (!VALID_CHAR_RE.test(family.name)) {
+							ctx.addIssue({
+								code: z.ZodIssueCode.custom,
+								message: `Family name "${family.name}" contains invalid characters for CSS variable generation. Specify the **as** property, read more at TODO:`,
+								path: ['families', i, 'name'],
+							});
+						}
+
+						// Check for name/as conflicts
+						const name = getFamilyName(family);
+						const key = 'as' in family ? 'as' : 'name';
+						const existing = visited.get(name);
+						if (existing) {
+							ctx.addIssue({
+								code: z.ZodIssueCode.custom,
+								message:
+									key === 'name' && existing === 'name'
+										? `Multiple families have the same **name** property: "${name}". Names have to be unique, you can override one of these family name by specifying the **as** property. Read more at TODO:`
+										: key === 'as' && existing === 'as'
+											? `Multiple families have the same **as** property: "${name}". Names have to be unique, update one of these family **as** property to avoid conflicts. Read more at TODO:`
+											: `A family **name** property is conflicting with another family **as** property: "${name}". Either update the current **as** property or add a unique **as** property to the family that has a **name** property. Read more at TODO:`,
```

Comment:
```suggestion
											: `A family **name** property is conflicting with another family **as** property: "${name}". All **name** and **as** values must be unique. Please rename to avoid conflicts. Read more at TODO:`,
```
I appreciate how specific your advice was there! This might be a time where explaining out exactly what to do, when there are two choices, is less helpful than describing the condition to be met, maybe? Do you think there's something like the above that could be used?